src/Native/System.Security.Cryptography.Native/pal_ssl.cpp

```diff
@@ -390,3 +440,134 @@ extern "C" int32_t SslRead(SSL* ssl, void* buf, int32_t num)
 {
     return SSL_read(ssl, buf, num);
 }
+
+extern "C" int32_t IsSslRenegotiatePending(SSL* ssl)
+{
+    return SSL_renegotiate_pending(ssl) != 0;
+}
+
+extern "C" int32_t SslShutdown(SSL* ssl)
+{
+    return SSL_shutdown(ssl);
+}
+
+extern "C" void SslSetBio(SSL* ssl, BIO* rbio, BIO* wbio)
+{
+    SSL_set_bio(ssl, rbio, wbio);
+}
+
+extern "C" int32_t SslDoHandshake(SSL* ssl)
+{
+    return SSL_do_handshake(ssl);
+}
+
+extern "C" int32_t IsSslStateOK(SSL* ssl)
+{
+    return SSL_state(ssl) == SSL_ST_OK;
+}
+
+extern "C" X509* SslGetPeerCertificate(SSL* ssl)
+{
+    return SSL_get_peer_certificate(ssl);
+}
+
+extern "C" X509Stack* SslGetPeerCertChain(SSL* ssl)
+{
+    return SSL_get_peer_cert_chain(ssl);
+}
+
+extern "C" int32_t SslCtxUseCertificate(SSL_CTX* ctx, X509* x)
+{
+    return SSL_CTX_use_certificate(ctx, x);
+}
+
+extern "C" int32_t SslCtxUsePrivateKey(SSL_CTX* ctx, EVP_PKEY* pkey)
+{
+    return SSL_CTX_use_PrivateKey(ctx, pkey);
+}
+
+extern "C" int32_t SslCtxCheckPrivateKey(SSL_CTX* ctx)
+{
+    return SSL_CTX_check_private_key(ctx);
+}
+
+extern "C" int32_t BioCtrlPending(BIO* bio)
```

Comment:
Since this doesn't use any SSL types (`SSL`, `SSL_CTX`, `PAL_SSL.*`, etc) I'd put it in pal_bio, personally.